CONTRIBUTING.md

```diff
@@ -1,4 +1,51 @@
 # Contributing
+## App Architecture
+This section describes how the application is structured, mainly the Python backend. This does not attempt to cover any information about the React frontend. As of this writing, the app is currently under a major refactoring to make the application more robust. There is code that uses the deprecated [RequestParser](https://flask-restful.readthedocs.io/en/latest/reqparse.html) which is being removed from the application in favor of [Marshmallow](https://marshmallow.readthedocs.io/en/stable/). It is expected that all new functionality uses Marshmallow.
+
+This application uses the [Flask](https://palletsprojects.com/p/flask/) micro-framework backed by an [sqlLite](https://sqlite.org/index.html) database ([Postgres](https://www.postgresql.org/) for production) utilizing [SQLAlchemy](https://docs.sqlalchemy.org/en/14/) as the ORM.
+ - [Flask-Restful](https://flask-restful.readthedocs.io/en/latest/) is used for routing to encourage Restful routes/resources.
+ - [Marshmallow](https://marshmallow.readthedocs.io/en/stable/) is being used for input validation, serialization, and deserialization.
+ - Flask-mail is currently being used to send mail, and Jinja is used for templating the email messages. However, the mail library has been deprecated and there is an open issue to switch to another library.
+ - [Flake8](https://gitlab.com/pycqa/flake8) is currently installed for linting, but probably not being used by many contributors. However, we will most likely be using [black](https://github.com/psf/black) in the future or a combination of Flake8 for linting and black for formatting.
+ - For database migrations, we recently installed [alembic](https://alembic.sqlalchemy.org/en/latest/) and will be using that in the future. For now, while the app is under development, we drop the DB and rebuild whenever there is a change.
+ - There are three different environments the application has been configured to run in: development, testing, and production.
+ - For testing, we are using [Pytest](https://docs.pytest.org/en/latest/).
+ - It is an expectation that all new functionality, changes in behavior, or bug fixes will be covered under **automated tests**.
+
+There are currently three main areas of the application to be familiar with, and possibly four in the future (dependent on how serialization will be handled).
+
+The three main areas are resources, models, and schemas. This is in line with an MVC web application. The resources directory contains all of the controllers/endpoints. Resources is the terminology that Flask-restful uses and it is the terminology used here. Models are the models, that is where the database tables are defined, and methods that are related to fetching data from the database. The Schemas are for the Marshmallow library, and they define how to validate, serialize, and deserialize the data.
```

Comment:
```suggestion
The three main areas of the application to be familiar with are resources, models, and schemas. This is in line with a Model-View-Controller (MVC) web application. 
```

Reply:
I like how concise your suggestion is. And it raises an issue that I hadn't thought of. When I read this paragraph I infer that a schema could be thought of the view in the MVC architecture, when in reality it is not. Since the backend is sending a JSON response and not rendering any views then we technically don't have the V in the MVC.

Perhaps it is better to say there are two main areas of the app (resources and models)?

Reply:
Perhaps:

"The three main areas of the application to be familiar with are resources, models, and schemas. The models and resources align respectively with the model and controller components of a Model-View-Controller (MVC) web application. Schemas are used for input validation and deserialization."

I think it's better to stick with three because the schemas sections is below this introduction. People might wonder if it's really important if it's not listed as one of the major areas list at the top of the section.

Reply:
I like that!